control-plane/csi-driver/src/bin/controller/client.rs

```diff
@@ -498,4 +499,25 @@ impl RestApiClient {
             .await?;
         Ok(response.into_body())
     }
+
+    /// Expand volume.
+    #[instrument(fields(volume.uuid = %volume_id, volume.size = %target_volume_size), skip(self, volume_id, target_volume_size))]
+    pub(crate) async fn expand_volume(
+        &self,
+        volume_id: &uuid::Uuid,
+        target_volume_size: u64,
+    ) -> Result<Volume, ApiClientError> {
+        let vol = self
+            .rest_client
+            .volumes_api()
+            // This call should be idempotent and should return success if the volume
```

Comment:
w.r.t csi it must be idempotent. I don't think REST is gonna get an error today as well if the size is already what we request.
@niladrih Can we confirm if **we get the CSI RPC at all** in case PVC is modified to a size that's same or lesser than current size?

Reply:
@tiagolobocastro Shall we throw back error to REST in such case?

Reply:
@dsharma-dc -- We should get the CSI call for changes to the PVC capacity (https://github.com/kubernetes-csi/external-resizer/blob/v1.10.0/pkg/controller/expand_and_recover.go#L160)
The spec requires that we be idempotent, so we ought to have idempotency here irrespective of what kubernetes descides to do here: https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#controllerexpandvolume

Reply:
I've added a check. I'm GET-ing the volume and comparing the size against the requested size.

Reply:
WDYT @tiagolobocastro @dsharma-dc ?